test/unit/buffer_spec.lua

```diff
@@ -211,93 +210,246 @@ describe('buffer functions', function()
   end)
 
   describe('build_stl_str_hl', function()
+    local buffer_byte_size = 100
+    local STL_MAX_ITEM = 80
+    local output_buffer = ''
+
+    -- This function builds the statusline
+    --
+    -- @param arg Optional arguments are:
+    --    .pat The statusline format string
+    --    .fillchar The fill character used in the statusline
+    --    .maximum_cell_count The number of cells available in the statusline
+    local build_stl_str_hl = function (arg)
```

Comment:
Declaring a function this way is not recommended, see https://github.com/mpeterv/luacheck/issues/60